stdlib/REPL/src/LineEdit.jl

```diff
@@ -395,16 +397,24 @@ function refresh_multi_line(termbuf::TerminalBuffer, terminal::UnixTerminal, buf
     # Now go through the buffer line by line
     seek(buf, 0)
     moreinput = true # add a blank line if there is a trailing newline on the last line
+    lastline = false # indicates when to stop printing lines, even when there are potentially
+                     # more (for the case where rows is too small to print everything)
+                     # Note: when there are too many lines for rows, we still print the first lines
+                     # even if they are going to not be visible in the end: for simplicity, but
+                     # also because it does the 'right thing' when the window is resized
     while moreinput
-        l = readline(buf, keep=true)
-        moreinput = endswith(l, "\n")
+        line = readline(buf, keep=true)
+        moreinput = endswith(line, "\n")
+        if moreinput && lastline # we want to print only one "visual" line, so
+            line = line[1:end-1] # don't include the trailing "\n"
+        end
         # We need to deal with on-screen characters, so use textwidth to compute occupied columns
-        llength = textwidth(l)
-        slength = sizeof(l)
+        llength = textwidth(line)
+        slength = sizeof(line)
         cur_row += 1
         # lwrite: what will be written to termbuf
-        lwrite = region_active ? highlight_region(l, regstart, regstop, written, slength) :
-                                 l
```

Comment:
I literally spent 5 minutes trying to find where was the else clause, thinking there was a bug and even going through git-log to find where this was introduced... So I allowed myself to make this unrelated change for clarity and maybe save time to future readers of this code.